File: Examples/test-suite/csharp/rev_len_str_runme.cs
using System;
using rev_len_strNamespace;

public class rev_len_str_runme {

  public static void Main() {
    Test t = new Test();
    string hile = "hile";
    string hil0 = "hil\0";
    if (t.strlen(hile) != 4)
      throw new Exception("bad multi-arg typemap");

    if (t.strlen(hil0) != 4)
      throw new Exception("bad multi-arg typemap");
  }
}

File: Examples/test-suite/lua/rev_len_str_runme.lua
require("import")	-- the import fn
import("rev_len_str")	-- import lib
v = rev_len_str

-- catch "undefined" global variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})

t = v.Test()
assert(t:strlen("hile") == 4, "bad multi-arg typemap");
assert(t:strlen("hil\0") == 4, "bad multi-arg typemap");
